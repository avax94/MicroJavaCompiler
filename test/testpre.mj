program Test14
{
	void main() int i[]; char c; {
	  c = chr(65);
      i = new int[ord(c)];	  
	  print(len(i));
	}
}