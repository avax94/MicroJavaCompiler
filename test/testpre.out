=====================SEMANTICKA ANALIZA=========================
Pretraga na (4), nadjeno Var c: char, 1, 1 
Pretraga na (4), nadjeno Meth chr: char, 0, 1 
Pretraga na (5), nadjeno Var i: Arr of int, 0, 1 
Pretraga na (5), nadjeno Meth ord: int, 0, 1 
Pretraga na (5), nadjeno Var c: char, 1, 1 
Pretraga na (6), nadjeno Meth len: int, 0, 1 
Pretraga na (6), nadjeno Var i: Arr of int, 0, 1 
=====================SINTAKSNA ANALIZA=========================
Pronadjeno: 0 globalnih promenljivih
Pronadjeno: 0 globalnih konstanti
Pronadjeno: 0 globalnih nizova
Pronadjeno: 2 lokalnih promenljivih
Pronadjeno: 1 globalnih funkcija
Pronadjeno: 0 definicija klasa
Pronadjeno: 0 metoda
Pronadjeno: 0 statement blokova
Pronadjeno: 4 poziva funkcija u mainu
Pronadjeno: 0 formalnih argumenata
Pronadjeno: 0 polja klasa
=====================SYMBOL TABLE DUMP=========================
Var $helper: int, 0, 0 
Prog Test14: notype, 0, 0 

Meth main: notype, 0, 0 
   Var i: Arr of int, 0, 1 
   Var c: char, 1, 1 

Type int: int, -1, -1 
Type char: char, -1, -1 
Con eol: char, 10, 0 
Con null: Class [], 0, 0 
Meth chr: char, 0, 1 
   Var i: int, 0, 1 

Meth ord: int, 0, 1 
   Var ch: char, 0, 1 

Meth len: int, 0, 1 
   Var arr: Arr of notype, 0, 1 

Type bool: , -1, -1 

************************USPESNO ZAVRSENO PARSIRANJE!*************************
